job-server-api/src/spark.jobserver/NamedRddSupport.scala

```diff
@@ -1,120 +1,206 @@
+      case NamedRDD(rdd, _, _) =>
+        rdd.unpersist(blocking = false)
+    }
+  }
+}
 
-  /**
-   * Replaces an existing RDD with a given name with a new RDD. If an old RDD for the given name existed,
-   * it is un-persisted (non-blocking) and destroyed. It is safe to call this method when there is no
-   * existing RDD with the given name. If multiple threads call this around the same time, the end result
-   * is undefined - one of the generated RDDs will win and will be returned from future calls to get().
-   *
-   * The rdd generator function will be called from the caller's thread. Note that if this is called at the
-   * same time as getOrElseCreate() for the same name, and completes before the getOrElseCreate() call,
-   * then threads waiting for the result of getOrElseCreate() will unblock with the result of this
-   * update() call. When the getOrElseCreate() succeeds, it will replace the result of this update() call.
-   *
-   * @param name the unique name of the RDD. The uniqueness is scoped to the current SparkContext.
-   * @param rddGen a 0-ary function which will be called to generate the RDD in the caller's thread.
-   * @param forceComputation if true, forces the RDD to be computed by calling count().
-   * @param storageLevel the storage level to persist the RDD with. Default: StorageLevel.MEMORY_ONLY.
-   * @tparam T the generic type of the RDD.
-   * @return the RDD with the given name.
-   */
-  def update[T](name: String,
-                rddGen: => RDD[T],
-                forceComputation: Boolean = true,
-                storageLevel: StorageLevel = defaultStorageLevel): RDD[T]
+/**
+ * @note please use NamedObjectSupport instead !
+ */
+trait NamedRddSupport extends NamedObjectSupport { self: SparkJob =>
 
   /**
-   * Destroys an RDD with the given name, if one existed. Has no effect if no RDD with this name exists.
+   * NamedRdds - a trait that gives you safe, concurrent creation and access to named RDDs
+   * (the native SparkContext interface only has access to RDDs by numbers).
+   * It facilitates easy sharing of RDDs amongst jobs sharing the same SparkContext.
+   * If two jobs simultaneously tries to create an RDD with the same name,
+   * only one will win and the other will retrieve the same one.
    *
-   * @param name the unique name of the RDD. The uniqueness is scoped to the current SparkContext.
+   * Note that to take advantage of NamedRddSupport, a job must mix this in and use the APIs here instead of
+   * the native RDD `cache()`, otherwise we will not know about the names.
    */
-  def destroy(name: String): Unit
+  trait _NamedRdds {
 
-  /**
-   * Returns the names of all named RDDs that are managed by the RddManager.
-   *
-   * Note: this returns a snapshot of RDD names at one point in time. The caller should always expect
-   * that the data returned from this method may be stale and incorrect.
+    val defaultTimeout = Timeout(Duration(60, java.util.concurrent.TimeUnit.SECONDS))
```

Comment:
You can just define something like `def config` as an abstract variable in the Trait  to be able to access the config, then in the implementation pass it in the constructor.

> On Jan 19, 2016, at 3:05 AM, TimMaltGermany <notifications@github.com> wrote:
> 
> In job-server-api/src/spark.jobserver/NamedRddSupport.scala <https://github.com/spark-jobserver/spark-jobserver/pull/350#discussion_r50099393>:
> 
> >  
> > -  /**
> > -   * Returns the names of all named RDDs that are managed by the RddManager.
> > -   *
> > -   * Note: this returns a snapshot of RDD names at one point in time. The caller should always expect
> > -   * that the data returned from this method may be stale and incorrect.
> > +    val defaultTimeout = Timeout(Duration(60, java.util.concurrent.TimeUnit.SECONDS))
> Still using the hard-coded timeout here for backwards compatibility and since the trait cannot access the config.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/spark-jobserver/spark-jobserver/pull/350/files#r50099393>.
>